process/block/baseProcess.go

```diff
@@ -1379,3 +1400,34 @@ func (bp *baseProcessor) ProcessScheduledBlock(header data.HeaderHandler, _ data
 
 	return err
 }
+
+func (bp *baseProcessor) getMarshalizedScheduledSCRs(mapSCRs map[block.Type][]data.TransactionHandler) ([]byte, error) {
+	scrsBatch := &batch.Batch{}
+	for blockType, scrs := range mapSCRs {
+		if len(scrs) == 0 {
+			continue
+		}
+
+		scheduledSCRs := &process.ScheduledSCRs{
+			BlockType:  blockType,
+			TxHandlers: make([]data.TransactionHandler, len(scrs)),
+		}
+
+		for scrIndex, scr := range scrs {
+			scheduledSCRs.TxHandlers[scrIndex] = scr
+		}
+
+		marshalizedScheduledSCRs, err := bp.marshalizer.Marshal(scheduledSCRs)
+		if err != nil {
+			return nil, err
+		}
+
+		scrsBatch.Data = append(scrsBatch.Data, marshalizedScheduledSCRs)
+	}
+
+	if len(scrsBatch.Data) == 0 {
+		return nil, nil
```

Comment:
Yep, it is used this way also in other places just to avoid to put empty stuff marshalized in storage. I will replace with make([]byte, 0) so this way the len check will remain 0 and we will keep the same desired functionality in the caller side